### PR TITLE
Feature/danish accident travel

### PR DESCRIPTION
**Pull request by meletios**
Add Accident and Travel for DK

### CODE REVIEW

File: src/main/kotlin/com/hedvig/underwriter/model/AddressData.kt [hunk 1]
```diff
@@ -4,5 +4,4 @@ interface AddressData {
     val street: String?
     val zipCode: String?
     val city: String?
-    val livingSpace: Int?
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/model/AddressData.kt:7]: Haha yeah why was this apart of AddressData and it was probably me :slightly_smiling_face:

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt [hunk 2]
```diff
@@ -400,7 +402,9 @@ class SignServiceImpl(
                     )
                     is NorwegianHomeContentsData,
                     is NorwegianTravelData -> QuotesSignData.NorwegianBankId(quotes[0].memberId!!, quotes[0].ssn)
-                    is DanishHomeContentsData -> QuotesSignData.DanishBankId(quotes[0].memberId!!, quotes[0].ssn)
+                    is DanishHomeContentsData,
+                    is DanishAccidentData,
+                    is DanishTravelData -> QuotesSignData.DanishBankId(quotes[0].memberId!!, quotes[0].ssn)
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:407]: I dont think you should do this change if it is 1 you should only be able to sign `DanishHomeContentsData` right? Then we should ad the possibility on 2 and 3. Let me know if this is unclear :slightly_smiling_face:

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/notificationService/NotificationServiceImpl.kt [hunk 3]
```diff
@@ -58,7 +64,18 @@ class NotificationServiceImpl(
                         true -> "YOUTH"
                         false -> "REGULAR"
                     }
-                    is DanishHomeContentsData -> "REGULAR"
+                    is DanishHomeContentsData -> when (quote.data.isStudent) {
+                        true -> "STUDENT"
+                        false -> "REGULAR"
+                    }
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/notificationService/NotificationServiceImpl.kt:70]: Good catch

File: src/main/resources/db/changelog/migrations/20201117-1030-add-danish-accident-and-travel-table.sql [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+(
+    internal_id serial PRIMARY KEY,
+    id          uuid NOT NULL
+);
+
+ALTER TABLE quote_revisions
+    ADD COLUMN quote_danish_accident_data_id integer NULL REFERENCES quote_revision_danish_accident_data (internal_id),
+    ADD COLUMN quote_danish_travel_data_id   integer NULL REFERENCES quote_revision_danish_travel_data (internal_id),
+    DROP CONSTRAINT single_quote_type,
+    ADD CONSTRAINT single_quote_type CHECK (
```
Comment (FredrikAreschoug) [inline src/main/resources/db/changelog/migrations/20201117-1030-add-danish-accident-and-travel-table.sql:21]: Starting to love this constraint :)

**Review by FredrikAreschoug — COMMENTED** [review]
A world of fun this! :tada: but well done. One important comment on Signing :slightly_smiling_face:

File: src/main/resources/db/changelog/migrations/20201117-1030-add-danish-accident-and-travel-table.sql (hunk 4 above)
Comment (meletios) [reply src/main/resources/db/changelog/migrations/20201117-1030-add-danish-accident-and-travel-table.sql:21]: HAHAHAHA Can we please burn down underwriter???

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt (hunk 2 above)
Comment (meletios) [reply src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:407]: Yes, i know what you mean, i will create a custom exception there

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt [hunk 5]
```diff
@@ -402,33 +402,71 @@ class SignServiceImpl(
...
-                var ssn: String? = null
-                quotes.forEach { quote ->
-                    if (quote.data is PersonPolicyHolder<*>) {
-                        quote.data.ssn?.let {
-                            ssn = it
-                            return@forEach
-                        }
-                    } else {
-                        throw RuntimeException("Quote data should not be able to be of type ${quote.data::class}")
+                    is DanishTravelData -> {
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:407]:
```suggestion
                   is DanishAccidentData || is DanishTravelData -> {
```

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt [hunk 6]
```diff
@@ -402,33 +402,71 @@ class SignServiceImpl(
...
+                    return@forEach
+                }
+            } else {
+                throw RuntimeException("Quote data should not be able to be of type ${quote.data::class}")
+            }
+        }
+        return ssn!!
+    }
+
+    private fun canTwoNorwegianBundledQuotesBeSigned(quotes: List<Quote>): Boolean =
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:453]:
```suggestion
    private fun isTwoNorwegianQuotes(quotes: List<Quote>): Boolean =
```

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt [hunk 7]
```diff
@@ -402,33 +402,71 @@ class SignServiceImpl(
...
+        quotes.size == 2 &&
+            quotes.any { quote -> quote.data is NorwegianHomeContentsData } &&
+            quotes.any { quote -> quote.data is NorwegianTravelData }
+
+    private fun canTwoDanishBundledQuotesBeSigned(quotes: List<Quote>): Boolean =
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:458]:
```suggestion
    private fun isTwoDanishQuotes(quotes: List<Quote>): Boolean =
```

File: src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt [hunk 8]
```diff
@@ -402,33 +402,71 @@ class SignServiceImpl(
...
+        quotes.size == 2 &&
+            (quotes.any { quote -> quote.data is DanishHomeContentsData } &&
+                (quotes.any { quote -> quote.data is DanishAccidentData } ||
+                    quotes.any { quote -> quote.data is DanishTravelData }))
+
+    private fun canThreeDanishBundledQuotesBeSigned(quotes: List<Quote>): Boolean =
```
Comment (FredrikAreschoug) [inline src/main/kotlin/com/hedvig/underwriter/service/SignServiceImpl.kt:464]:
```suggestion
    private fun isThreeDanishQuotes(quotes: List<Quote>): Boolean =
```

**Review by FredrikAreschoug — APPROVED** [review]
Looks good! Log error on accident as well :) and some minor naming

**Review by FredrikAreschoug — DISMISSED** [review]
yay :tada:

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/productPricing/dtos/InsuranceType.kt [hunk 9]
```diff
@@ -10,5 +10,10 @@ enum class InsuranceType {
     NORWEGIAN_TRAVEL,
     NORWEGIAN_YOUTH_HOME_CONTENTS,
     NORWEGIAN_YOUTH_TRAVEL,
-    DANISH_HOME_CONTENTS
+    DANISH_HOME_CONTENTS,
```
Comment (AlexSJ1990) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/productPricing/dtos/InsuranceType.kt:13]: could we use product-pricing-objects lib here?

File: src/main/kotlin/com/hedvig/underwriter/serviceIntegration/productPricing/dtos/contract/AgreementQuote.kt [hunk 10]
```diff
@@ -98,6 +100,28 @@ sealed class AgreementQuote {
         val lineOfBusiness: DanishHomeContentLineOfBusiness
     ) : AgreementQuote()
 
+    data class DanishAccidentQuote(
```
Comment (AlexSJ1990) [inline src/main/kotlin/com/hedvig/underwriter/serviceIntegration/productPricing/dtos/contract/AgreementQuote.kt:103]: same question here

**Review by AlexSJ1990 — DISMISSED** [review]
Looks really good! 🎉 2 small comments

**Review by AlexSJ1990 — APPROVED** [review]
Nice! 🎉

**Pull request merged**